instrumentation/mongo/mongo-common/testing/src/main/java/io/opentelemetry/instrumentation/mongo/testing/AbstractMongoClientTest.java

```diff
@@ -0,0 +1,546 @@
+                                "{\"getMore\":\"?\",\"collection\":\"?\",\"batchSize\":\"?\",\"$db\":\"?\"}",
+                                "{\"getMore\":\"?\",\"collection\":\"?\",\"batchSize\":\"?\",\"$db\":\"?\",\"$readPreference\":{\"mode\":\"?\"}}",
+                                "{\"getMore\":\"?\",\"collection\":\"?\",\"batchSize\":\"?\",\"$db\":\"?\",\"lsid\":{\"id\":\"?\"}}"))));
+  }
+
+  @Test
+  @DisplayName("test error")
+  void testError() {
+    assertThatIllegalArgumentException().isThrownBy(() -> error("test_db", createCollectionName()));
+    // Unfortunately not caught by our instrumentation.
+    assertThat(testing().spans().size()).isEqualTo(0);
+  }
+
+  @Test
+  @DisplayName("test create collection with already built ClientOptions")
+  void testCreateCollectionWithAlreadyBuiltClientOptions() {
+    String dbName = "test_db";
+    String collectionName = createCollectionName();
+
+    testing()
+        .runWithSpan(
+            "parent", () -> createCollectionWithAlreadyBuiltClientOptions(dbName, collectionName));
+
+    testing()
+        .waitAndAssertTraces(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                    span ->
+                        mongoSpan(
+                            trace,
+                            1,
+                            "create",
+                            collectionName,
+                            dbName,
+                            trace.getSpan(0),
+                            asList(
+                                "{\"create\":\"" + collectionName + "\",\"capped\":\"?\"}",
+                                "{\"create\":\""
+                                    + collectionName
+                                    + "\",\"capped\":\"?\",\"$db\":\"?\",\"$readPreference\":{\"mode\":\"?\"}}"))));
+  }
+
+  protected String createCollectionName() {
+    return "testCollection-" + collectionIndex.getAndIncrement();
+  }
+
+  @SuppressWarnings("deprecation")
+  // TODO DbIncubatingAttributes.DB_CONNECTION_STRING deprecation
+  void mongoSpan(
+      TraceAssert trace,
+      int index,
+      String operation,
+      String collection,
+      String dbName,
+      Object parentSpan,
+      List<String> statements) {
+    SpanDataAssert span =
+        trace.element(index).hasName(operation + " " + dbName + "." + collection).hasKind(CLIENT);
+    if (parentSpan == null) {
+      span.hasNoParent();
+    } else {
+      span.hasParent((SpanData) parentSpan);
+    }
+
+    span.hasAttributesSatisfyingExactly(
+        equalTo(ServerAttributes.SERVER_ADDRESS, host),
+        equalTo(ServerAttributes.SERVER_PORT, port),
+        satisfies(
+            DbIncubatingAttributes.DB_STATEMENT,
+            val -> val.matches(statement -> statements.contains(statement.replaceAll(" ", "")))),
```

Comment:
I don't like this much because the error messages it produces aren't too useful. For example on failure it will produce
```
[STRING attribute 'db.statement'] 
Expecting actual:
  "a{ "count" : "testCollection-3", "query" : { } }"
to match given predicate.
```
I'd change it to
```
        satisfies(
            DbIncubatingAttributes.DB_STATEMENT,
            val -> val.satisfies(statement -> assertThat(statements).contains(statement.replaceAll(" ", "")))),
``` 
which produces
```
[STRING attribute 'db.statement'] (1 failure)
-- failure 1 --
Expecting ArrayList:
  ["{"count":"testCollection-3","query":{}}",
    "{"count":"testCollection-3","query":{},"$db":"?"}",
    "{"count":"testCollection-3","query":{},"$db":"?","lsid":{"id":"?"}}",
    "{"count":"testCollection-3","query":{},"$db":"?","$readPreference":{"mode":"?"}}",
    "{"count":"testCollection-3","$db":"?","lsid":{"id":"?"}}"]
to contain:
  ["a{"count":"testCollection-3","query":{}}"]
but could not find the following element(s):
  ["a{"count":"testCollection-3","query":{}}"]
```